platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1027,7 +1026,10 @@ bool OfflineDatabase::markUsed(int64_t regionID, const Resource& resource) {
         selectQuery.bind(4, tile.x);
         selectQuery.bind(5, tile.y);
         selectQuery.bind(6, tile.z);
-        return !selectQuery.run();
+
+        bool notOnOtherRegion = !selectQuery.run();
+
+        return notOnThisRegion && notOnOtherRegion;
```

Comment:
return statement is &&, why do we need to run two queries, maybe return in each branch? Sorry if I missed some important bit.

Reply:
Discussed offline, it is correct, both queries has to be run to check that resource is missing in both regions.